test/e2e/agent/agent_test.go

```diff
@@ -63,4 +65,207 @@ var _ = Describe("VM Agent behavior", func() {
+				logrus.Infof("Updating %s with config %s", deviceId, device.Spec.Config)
+			})
+
+			// Check the http config error is detected.
+			harness.WaitForDeviceContents(deviceId, `1 invalid configuration: example-git-config-provider. Error: failed fetching
+			specified Repository definition 00000000-0000-0000-0000-000000000000/not-existing-repo:
+			resource not found`,
+				func(device *v1alpha1.Device) bool {
+					return conditionExists(device, "SpecValid", "False", "Invalid")
+				}, "2m")
+
+			// The behaviour will change after EDM-418.
+			harness.WaitForDeviceContents(deviceId, "the device is updated to renderedVersion: 1",
+				func(device *v1alpha1.Device) bool {
+					return conditionExists(device, "Updating", "False", "Updated")
+				}, "2m")
+			Eventually(harness.GetDeviceWithStatusSummary, TIMEOUT, POLLING).WithArguments(
+				deviceId).Should(Equal(v1alpha1.DeviceSummaryStatusType("Online")))
+		})
+
+		It(`should show an error when trying to update a device with a httpConfigProviderSpec
+			with invalid Path, and report 'Online' status`, func() {
+			deviceId, _ := harness.EnrollAndWaitForOnlineStatus()
+
+			// Create the http repository.
+			_, err := model.NewRepositoryFromApiResource(&httpRepo)
+			Expect(err).ToNot(HaveOccurred())
+
+			// Update the device with the http invalid config.
+			harness.UpdateDeviceWithRetries(deviceId, func(device *v1alpha1.Device) {
+				logrus.Infof("current device is %s", deviceId)
+				// Create ConfigProviderSpec.
+				var configProviderSpec v1alpha1.ConfigProviderSpec
+				err := configProviderSpec.FromHttpConfigProviderSpec(httpConfigInvalidPath)
+				Expect(err).ToNot(HaveOccurred())
+
+				device.Spec.Config = &[]v1alpha1.ConfigProviderSpec{configProviderSpec}
+				logrus.Infof("updating %s with config %s", deviceId, device.Spec.Config)
+			})
+
+			// Check the http config error is detected.
+			harness.WaitForDeviceContents(deviceId, "1 invalid configuration: . Error: sending HTTP Request",
+				func(device *v1alpha1.Device) bool {
+					return conditionExists(device, "SpecValid", "False", "Invalid")
+				}, "2m")
+
+			// The behaviour will change after EDM-418.
+			harness.WaitForDeviceContents(deviceId, "the device is updated to renderedVersion: 1",
+				func(device *v1alpha1.Device) bool {
+					return conditionExists(device, "Updating", "False", "Updated")
+				}, "2m")
+			Eventually(harness.GetDeviceWithStatusSummary, TIMEOUT, POLLING).WithArguments(
+				deviceId).Should(Equal(v1alpha1.DeviceSummaryStatusType("Online")))
+		})
+
+		It("should report 'Unknown' after the device vm is powered-off", func() {
+			deviceId, _ := harness.EnrollAndWaitForOnlineStatus()
+
+			// Shutdown the vm.
+			err := harness.VM.Shutdown()
+			Expect(err).ToNot(HaveOccurred())
+			Eventually(harness.GetDeviceWithStatusSummary, LONGTIMEOUT, POLLING).WithArguments(
+				deviceId).Should(Equal(v1alpha1.DeviceSummaryStatusType("Unknown")))
+		})
+	})
 })
+
+var mode = 0644
```

Comment:
I think I can't  make it a const, since in go consts are not addressable: https://golang.org/ref/spec#Address_operators and cannot be the operand of the address operator "&". I need to assign it to as a pointer to the inline config